src/registry_schemas/schemas/amalgamation.json

```diff
@@ -0,0 +1,106 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://bcrs.gov.bc.ca/.well_known/schemas/amalgamation",
+    "type": "object",
+    "definitions": {},
+    "title": "Amalgamation Application Schema",
+    "required": [
+        "amalgamation"
+    ],
+    "properties": {
+        "amalgamation": {
+            "type":"object",
+            "required": [
+                "type",
+                "amalgamatingBusinesses",
+                "nameRequest",
+                "offices",
+                "parties",
+                "contactPoint",
+                "courtApproval"
+            ],
+            "properties": {
+                "type": {
+                    "type": "string",
+                    "title": "The type of Amalgamation",
+                    "enum": [
+                        "regular",
+                        "vertical",
+                        "horizontal"
+                    ]
+                },
+                "amalgamatingBusinesses": {
+                    "type": "array",
+                    "items": {
+                        "oneOf": [
+                            {
+                                "required": [
+                                    "role",
+                                    "identifier"
+                                ]
+                            },
+                            {
+                                "required": [
+                                    "role",
+                                    "foreignJurisdiction",
+                                    "legalName",
+                                    "corpNumber"
+                                ]
+                            }
+                        ],
+                        "properties": {
+                            "role": {
+                                "type": "string",
+                                "enum": [
+                                    "holding",
+                                    "amalgamating"
+                                ]
+                            },
+                            "identifier": { "type": "string" },
+                            "foreignJurisdiction": { 
+                                "$ref": "https://bcrs.gov.bc.ca/.well_known/schemas/foreign_jurisdiction" 
+                            },
+                            "legalName": { "type": "string" },
+                            "corpNumber": { "type": "string" }
```

Comment:
do we want to prefix `legalName` and `corpNumber` with `foreign`?

Reply:
The UI doesn't need to -- it uses a union type like this, so the properties are kept separate by the language.

![image](https://github.com/bcgov/business-schemas/assets/10002889/8877dded-1ec4-43b8-9480-e6012258e875)

Reply:
I think it makes sense for the schema definition to prefix `legalName` and `corpNumber` with `foreign` for clarity though.

So it sounds like the FE will be fine if we change the naming of these two fields?

Reply:
A question, is there any possibility/need FE will use `legalName` for non-foreign companies as well? If so, `legalName` will be shared like `role`.

Reply:
No, UI will not use `legalName` except for foreign companies.

The UI would need to be updated to prefix those properties with `foreign` but it's under development right now so not big deal, but I still think it's not necessary -- we should use the typing tools at hand to distinguish object types.

Reply:
ok, @chenhongjing forget about the `foreign` prefix.

Reply:
Hahaha...